### ISSUE TITLE
/v0/markets with groupId returning duplicates, not returning all markets

### ISSUE THREAD

**Issue by adam-binks** [manifoldmarkets/manifold#2798]
With the following code for pagination, I get duplicates of the same markets over and over, and don't fetch all markets in the group.

Is this the right way to do pagination with `before`? Shouldn't there be an `after` instead?

```ts
async function fetchMultiple({ approxLimit }: { approxLimit: number }) {
  const PAGE_LIMIT = Math.min(1000, approxLimit)
  let lastMarketId: string | undefined = undefined
  let markets: LiteMarket[] = []
  let totalFetched = 0

  while (totalFetched < approxLimit) {
    const url = new URL("https://api.manifold.markets/v0/markets")
    url.searchParams.append("groupId", "yEWvvwFFIqzf8JklMewp") // AI
    url.searchParams.append("limit", PAGE_LIMIT.toString())
    if (lastMarketId) {
      url.searchParams.append("before", lastMarketId)
    }

    console.log("Fetching", url.toString())

    const response = await fetch(url.toString())
    if (!response.ok) {
      throw new Error(`Manifold API error! status: ${response.status}`)
    }
    const newMarkets: LiteMarket[] = await response.json()
    if (newMarkets.length === 0) {
      break
    }
    lastMarketId = newMarkets[newMarkets.length - 1].id
    markets.push(...newMarkets)
    totalFetched += newMarkets.length

    await new Promise((resolve) => setTimeout(resolve, 600))
  }
  console.log("Fetched", markets.length, "markets")
  return markets
}
```

It's possible there's a bug in the above.

This is used in https://theaidigest.org/timeline

**Comment by adam-binks** [comment]
The fix is that you now need to:
```ts
url.searchParams.append("order", "desc")
```

I think this was not needed until a recent change, because our existing code stopped working, so maybe worth documenting/fixing.

**Comment by jahooma** [comment]
Ah, gotcha. Thanks Adam!

It looks like we just accidentally switched the default sort to ascending, in
this commit.
<https://github.com/manifoldmarkets/manifold/commit/da886649a3b708767afdd635cbf8477b73c3f7c3>
I'll switch it back to descending!

Cheers,
James

**Comment by adam-binks** [comment]
Thanks James! :D

**Comment by sipec** [comment]
fixed by fdb292461eba2433b8d44899e6d914b428c00972
